CRM/Batch/BAO/Batch.php

```diff
@@ -594,6 +596,7 @@ public static function exportFinancialBatch($batchIds, $exportFormat) {
     else {
       CRM_Core_Error::fatal("Could not locate exporter: $exporterClass");
     }
+    $exporter->_isDownloadFile = $downloadFile;
```

Comment:
Have you tried it? I think this will give a fatal error trying to set a protected property.